### PR TITLE
Use redis to store IDs for uploaded FB videos to prevent dupes

### PR DESCRIPTION
**Pull request by danielbachhuber**
See #7

### CODE REVIEW

File: script.py [hunk 1]
```diff
@@ -57,17 +59,26 @@ def upload():
 cli.add_command(upload)
 
 
+def get_redis():
+    """
+    Get the working redis instance
+    """
+    return redis.StrictRedis(host='localhost', port=6379, db=0)
```
Comment (danielbachhuber) [inline script.py]: @mpatek What do you think the behavior should be if Redis isn't running? And what's the appropriate way to use default config variables, but allow overriding with `.env` ?

Comment (mpatek) [inline script.py]: > What do you think the behavior should be if Redis isn't running?

I suppose you could define getter/setter routines that just do nothing if Redis is not running. We are doing something similar with the `cache` decorator in `glance.cache`.

> what's the appropriate way to use default config variables, but allow overriding with .env

You can do `os.getenv('VAR_NAME', 'default value')`

**Comment by danielbachhuber** [comment]
@mpatek #reviewmerge?

File: script.py [hunk 2]
```diff
@@ -57,17 +60,48 @@ def upload():
 cli.add_command(upload)
 
 
+@memoize
+def get_redis():
+    """
+    Get the working redis instance
+    """
+    try:
+        r = redis.StrictRedis(host=os.getenv('REDIS_HOST', 'localhost'), port=os.getenv('REDIS_PORT', 6379), db=0)
```
Comment (mpatek) [inline script.py]: If you're going to be running this on Heroku with redistogo you may need to also specify `password=os.getenv('REDIS_PASSWORD')`

File: script.py [hunk 3]
```diff
@@ -57,17 +60,48 @@ def upload():
...
+    return r
+
+
+def get_value(key):
+    """
+    Get a value for a key in redis
+    """
+    r = get_redis()
+    if r is not False:
+        r.get(key)
```
Comment (mpatek) [inline script.py]: This should be `return r.get(key)` I think.

Comment (danielbachhuber) [inline script.py]: Derp 1e5892a

File: script.py (hunk 2 above)
Comment (danielbachhuber) [inline script.py]: 63367a5

**Comment by danielbachhuber** [comment]
@mpatek Addressed your comments.

**Pull request merged**